ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/event-handlers.j2

```diff
@@ -9,7 +9,40 @@
 !
 event-handler {{ handler.name }}
 {%         if handler.trigger is arista.avd.defined %}
+{%             if handler.trigger is arista.avd.defined("on-maintenance")
+                  and handler.trigger_on_maintenance is arista.avd.defined
+                  and handler.trigger_on_maintenance.operation is arista.avd.defined
+                  and handler.trigger_on_maintenance.stage is arista.avd.defined %}
+{%                 set trigger_on_manitenance_cli =  "trigger " ~ handler.trigger ~ " " ~ handler.trigger_on_maintenance.operation %}
+{%                 if handler.trigger_on_maintenance.bgp is arista.avd.defined %}
+{%                     set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ " bgp " %}
+{%                     if handler.trigger_on_maintenance.bgp.neighbor_address_ipv4 is arista.avd.defined %}
+{%                         set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ handler.trigger_on_maintenance.bgp.neighbor_address_ipv4 %}
+{%                     elif handler.trigger_on_maintenance.bgp.neighbor_address_ipv6 is arista.avd.defined %}
+{%                         set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ handler.trigger_on_maintenance.bgp.neighbor_address_ipv6 %}
+{%                     elif handler.trigger_on_maintenance.bgp.peer_group_name is arista.avd.defined %}
+{%                         set trigger_on_manitenance_cli = trigger_on_manitenance_cli ~ handler.trigger_on_maintenance.bgp.peer_group_name %}
+{%                     endif %}
+{%                 elif handler.trigger_on_maintenance.interface.interface_type is arista.avd.defined and handler.trigger_on_maintenance.interface.interface_number is arista.avd.defined %}
```

Comment:
a lot of these elif/else assumes there can be only one trigger. Please check the CLI if multiple conditions could generate separate lines of CLI.

Reply:
it is taking only one trigger event at a time from the events we have in our schema